drivers/usb/device/usb_dc_sam0.c

```diff
@@ -119,9 +119,13 @@ static void usb_sam0_isr(void)
 	/* Acknowledge all interrupts */
 	regs->INTFLAG.reg = intflag;
 
+	/* End of reset interrupt */
 	if ((intflag & USB_DEVICE_INTFLAG_EORST) != 0U) {
 		UsbDeviceEndpoint *endpoint = &regs->DeviceEndpoint[0];
 
+		/* Enable suspend interrupt */
+		regs->INTENSET.reg = USB_DEVICE_INTENSET_SUSPEND;
```

Comment:
Please recheck, resume (EORSM) has to be handled to (always suspend+resume). No need to enable/disable them conditionally.

Reply:
> Please recheck, resume (EORSM) has to be handled to (always suspend+resume).

This PR is solely intended to address the lack of any "disconnect" event handling, EORSM (and others) can be addressed in #24402.

> No need to enable/disable them conditionally.

As far as I can see, there is no need, but this is what ASF does, so I am going to follow what ASF does.

Reply:
>This PR is solely intended to address the lack of any "disconnect" event handling, which leads to USB transfers not being cancelled when disconnected; EORSM (and others) can be addressed in #24402.

But it is not just a detach event, the host (host conroler) can put the device to suspend (7.1.7.6 Suspending "Devices can go into the Suspend state from any powered state.").

Reply:
> But it is not just a detach event, the host (host conroler) can put the device to suspend (7.1.7.6 Suspending "Devices can go into the Suspend state from any powered state.").

As I previously mentioned, this PR only addresses handling of the suspend interrupt in the context of "detach"/"disconnect" event in order to fix the issue described in #26656.

Since SAM0 USB core implicitly handles (that is, automatically acknowledges) all disabled interrupts, handling wakeup and/or resume interrupts (note that there are two and you seem to be referring to the WAKEUP interrupt, not EORSM here) do not necessarily have to come in conjunction with handling of the suspend interrupt; hence, I consider that out of scope for this PR.

As I noted above in the PR description, we will eventually need to address the rest of the interrupts (WAKEUP, EORSM, SOF, ...) as well, but that is the story for another PR.

Once again, this PR is a critical bug fix for #26656, not a feature enhancement.

Reply:
Once again, SUSPEND is not a detach event and you have to implement RESUME too, this is **mandatory** for the stack and you patch does not fix the bug (or maybe it introduces a new one), the atsamr21-xpro for example still hard-faults after few suspend events from the host. That may work for you because you plug and unplug the cable and the host resets and re-enumerates the device. This will not go in how it is.

Reply:
> SUSPEND is not a detach event

well, it clearly functions like one on the SAM0 SoC -- do you know any other way to detect detach/disconnect event on it because I would really appreciate it if you could tell me?

> you patch does not fix the bug

I am starting to wonder if you actually took time to read this PR and #26656 because it clearly is a fix for the aforementioned (rather critical) bug.

> the atsamr21-xpro for example still hard-faults after few suspend events from the host.

That has always been the case and this PR is clearly an incremental fix to what is a critical issue that renders USB useless on SAM0 SoCs.

> This will not go in how it is.

My intention was to make this PR a bug fix for #26656, a not feature enhancement; but if you insist that this becomes one, then I would be more than happy to make it one.